designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala

```diff
@@ -70,6 +70,8 @@ trait DeploymentManager extends AutoCloseable {
 
   def customActionsDefinitions: List[CustomActionDefinition]
 
+  def scenarioActivityHandling: ScenarioActivityHandling
```

Comment:
Done based on discussions, but slightly different than discussed, with IMO good justification:

3 cases were analyzed:
- manager does not handle any additional activities
- manager stores its custom activities itself (periodic manager - has its own database)
- manager wants to save and modify activities, stored in the core Nu database (new case, in the future periodics may work that way)

But those 3 cases are about both what manager exposes, and what abilities it wants to have.

From the point of view of what the manager exposes we have only 2 cases:
- all activities handled by Nu, manager does not store any activities
- some manager-specific activities stored and returned by manager

And from point of view of what manager wants to have provided, we have only one case, applicable to all managers - we always want to give the ability to store activities in core Nu repository. We can provide that ability always, it doesn't matter that most managers will not use it.

So:
- when it comes to what the manager exposes, we have `def scenarioActivityHandling` (trait member, because the decorator pattern is used, and self-types are troublesome)
- when it comes to what abilities we provide to the manager, we always provide `ScenarioActivityManager` inside the `DeploymentManagerDependencies`, with abilities to store and edit activities.